sda-download/internal/config/config.go

```diff
@@ -380,9 +379,17 @@ func (c *Map) appConfig() error {
 	}
 
 	var err error
-	c.App.Crypt4GHPrivateKey, c.App.Crypt4GHPublicKeyB64, err = GenerateC4GHKey()
-	if err != nil {
-		return err
+	if viper.GetString("c4gh.transientKeyPath") != "" {
+
+		if !viper.IsSet("c4gh.transientPassphrase") {
+			return errors.New("c4gh.transientPassphrase is not set")
+		}
+
```

Comment:
```suggestion
	if viper.IsSet("c4gh.transientKeyPath") && viper.IsSet("c4gh.transientPassphrase") {
```

No need to do extra checks since the error will come from the `GetC4GHKeys` function anyways.

Reply:
yes, but it is preferable to fail there imo

Reply:
Clearly we have a different view on what's preferable.